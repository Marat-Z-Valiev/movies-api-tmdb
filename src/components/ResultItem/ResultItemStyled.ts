import styled from "styled-components";

const ResultItemStyled = styled.div`
	display: grid;
	justify-items: center;

	h2,
	h3 {
		text-align: center;
		margin: 0;
	}

	h3 {
		margin-bottom: 15px;
	}

	img {
		border: 2px solid black;
		justify-self: center;
		opacity: 1;
		transition: opacity 0.25s ease-in-out;

		&:hover {
			opacity: 0.7;
		}
	}
	}

	.no-image {
		width: 346px;
		height: 517px;
	}

	@media (min-width: 320px) and (max-width: 425px) {
		h3 {
			margin-bottom: 20px;
		}
		img {
			margin-bottom: 20px;
		}
	}
`;

export default ResultItemStyled;
