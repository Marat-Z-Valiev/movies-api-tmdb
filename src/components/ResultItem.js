import React from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";
import ResultItemStyled from "./styles/ResultItemStyled";
import noImage from "../images/no-image-available.jpg";

const ResultItem = ({result}) => {
	const {id, vote_average, title, release_date, poster_path} = result;

	return (
		<ResultItemStyled>
			<h2>{vote_average == 0 ? "" : `Popularity ${vote_average} / 10`}</h2>
			<h2>{title}</h2>
			<h3>{release_date}</h3>
			<Link to={`/movieId=${id}`}>
				{poster_path === null ? (
					<img className="no-image" src={noImage} alt="no image available" />
				) : (
					<img
						src={`https://image.tmdb.org/t/p/w342/${poster_path}`}
						alt={title}
					/>
				)}
			</Link>
		</ResultItemStyled>
	);
};

ResultItem.propTypes = {
	result: PropTypes.object.isRequired
};

export default ResultItem;
