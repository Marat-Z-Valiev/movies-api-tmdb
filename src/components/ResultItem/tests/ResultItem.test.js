import React from "react";
import {shallow, configure, mount} from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import ResultItem from "../ResultItem";

configure({adapter: new Adapter()});

describe("ResultItem", () => {
	const props = {
		result: {},
	};
	it("should render correctly", () => {
		const wrapper = shallow(<ResultItem {...props} noImage />);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});
});
