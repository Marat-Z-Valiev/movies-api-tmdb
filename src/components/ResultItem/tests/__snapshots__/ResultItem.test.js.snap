// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultItem should render correctly 1`] = `
<ResultItemStyled>
  <h2>
    Popularity undefined / 10
  </h2>
  <h2 />
  <h3 />
  <Link
    to="/movieId=undefined"
  >
    <img
      src="https://image.tmdb.org/t/p/w342/undefined"
    />
  </Link>
</ResultItemStyled>
`;
